src/couch_replicator/src/couch_replicator_utils.erl

```diff
@@ -276,6 +278,88 @@ seq_encode(Seq) ->
     % object. We are being maximally compatible here.
     ?JSON_ENCODE(Seq).
 
+%% Log uses of http protocol and uses of https protocol where verify_peer
+%% would fail.
+log_security_warnings(#rep{} = Rep) ->
+    case config:get_boolean("couch_replicator", "log_security_warnings", false) of
+        false ->
+            ok;
+        true ->
+            ok = check_security(Rep, source),
+            ok = check_security(Rep, target)
+    end.
+
+check_security(#rep{} = Rep, Type) ->
+    Url =
+        case Type of
+            source -> Rep#rep.source#httpdb.url;
+            target -> Rep#rep.target#httpdb.url
+        end,
+    #url{protocol = Protocol} = ibrowse_lib:parse_url(Url),
+    case Protocol of
+        http ->
+            couch_log:warning("**security warning** replication ~s has insecure ~s at ~s", [
+                rep_principal(Rep), Type, Url
+            ]),
+            ok;
+        https ->
+            CACertFile = config:get("couch_replicator", "cacertfile"),
+            case CACertFile of
+                undefined ->
+                    couch_log:warning("security warnings enabled but no cacertfile configured", []);
+                CACertFile ->
+                    try
+                        ibrowse:send_req(Url, [], head, [], [
```

Comment:
Here we'd be blocking the scheduler when a node boot as each job tries to make ibrowse calls. Also, in the replicator we typically use our own connection pool and don't rely on ibrowse for that.

It's good that we do a try...catch but some requests start timing out at a minute or so, it would still block for too long and cause the scheduler to misbehave.

Reply:
moved invocation to the parser side as suggested.

Reply:
Ideally we would pass in `dbname` to parser so it can both update the `#rep` record with it, and log http warnings. But that's a bit of a larger change. So, checking the protocols schema right after the parsing step, like done here, will work well. 

However, since the TLS certificate check makes an external http connection we don't want to do that from any singleton replicator process. That includes the db event handler, the design doc processor, or the scheduler. A better place would seem to be the job process itself after it started and initialized. Somewhere in https://github.com/apache/couchdb/blob/main/src/couch_replicator/src/couch_replicator_scheduler_job.erl#L177 perhaps.

In the replication job, to improve isolation and not block the process if the verify function misbehaves, we could spawn separate TLS cert checker process for source and target endpoints. The would be spawn_link-ed from https://github.com/apache/couchdb/blob/main/src/couch_replicator/src/couch_replicator_scheduler_job.erl#L182 somewhere, then added to `#rep_state{}` record. After performing their `head` requests, emitting warning logs, they would exit. When they die we'd note their exit and clean them up in the `handle_info({'EXIT', ...})` handlers similar to https://github.com/apache/couchdb/blob/main/src/couch_replicator/src/couch_replicator_scheduler_job.erl#L270-L272. Because they are linked to the main process if the main replication job crashes, they would be automatically cleaned up as well.